### PR TITLE
Add an option to run MOM6 with ocean biogeochemistry (BLING model)

### LINKED ISSUE #17: Add an option to run MOM6 with ocean biogeochemistry (BLING model)
## Description

Add an option to build and run MOM6 with ocean biogeochemistry (BLING model). This is a preparation step for ocean color data assimilation development and activities.

## Solution

- in soca-bundle CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to git clone ocean_bgc repo recursively.

- in ocean_bgc CMakelist, export a list of ocean_bgc_src_files.

- in mom6 CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to import ocean_bgc_src_files and build them into mom6 libraries and executable.

- in soca, include BLING tracers in the MOM.res.nc restart file, use MOM_override to activate generic tracer packages, add an input_bgc.nml to set up ocean_bgc options (e.g. use BLING or other models), then add an optional new ctest forecast_mom6_bgc.test (if ENABLE_OCEAN_BGC = ON).

### LINKED ISSUE #394: Add an option to run MOM6 with ocean biogeochemistry (BLING model)
## Description

Add an option to build and run MOM6 with ocean biogeochemistry (BLING model). This is a preparation step for ocean color data assimilation development and activities.

## Solution

- in soca-bundle CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to git clone ocean_bgc repo recursively.

- in ocean_bgc CMakelist, export a list of ocean_bgc_src_files.

- in mom6 CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to import ocean_bgc_src_files and build them into mom6 libraries and executable.

- in soca, include BLING tracers in the MOM.res.nc restart file, use MOM_override to activate generic tracer packages, add an input_bgc.nml to set up ocean_bgc options (e.g. use BLING or other models), then add an optional new ctest forecast_mom6_bgc.test (if ENABLE_OCEAN_BGC = ON).

### PR DESCRIPTION
**Pull request by liuxiao37k**
## Description

This is one of the two PRs (in mom6 and soca, respectively) in order to add an option to build and run MOM6 with ocean biogeochemistry (BLING model). To do so, the following steps are needed:

- in ocean_bgc CMakelist, export a list of ocean_bgc_src_files (already merged to the ocean_bgc repo).

- add ocean_bgc as a submodule (pointer to the repo) in mom6 under /pkg.

- in mom6 CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to import ocean_bgc_src_files and build them into mom6 libraries and executable.

- in soca (this PR), include BLING tracers in the MOM.res.nc restart file, use MOM_override to activate generic tracer packages, append input.nml to set up ocean_bgc options (e.g. use BLING or other models), then add an optional new ctest forecast_mom6_bgc.test (if ENABLE_OCEAN_BGC = ON) to run MOM6 with BLING.

### Issue(s) addressed

- fixes #394 
- fixes JCSDA/MOM6/issues/17

## Testing

The three PRs were tested together on random Hera computing nodes:
ecbuild --build=release -DMPIEXEC_EXECUTABLE=`which srun` -DMPIEXEC_NUMPROC_FLAG="-n" -DBUILD_ECKIT=ON -DBUILD_CRTM=OFF -DENABLE_OCEAN_BGC=ON (soca_bundle_dir)

All 48 ctests (47 if -DENABLE_OCEAN_BGC=OFF) passed.

## Dependencies

- waiting on JCSDA/MOM6/pull/18

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
@liuxiao37k , your added test is not triggered in travis-ci. You need to add **-DENABLE_OCEAN_BGC=ON** to [line 29 of .travis.yml](https://github.com/JCSDA/soca/blob/3609e63295e52fb6cbef1f9db54de20270a600f9/.travis.yml#L29)

File: test/Data/72x35x25/input_bgc.nml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+ &MOM_input_nml
```
Comment (guillaumevernieres) [inline test/Data/72x35x25/input_bgc.nml]: Couldn't we just append what's needed to `input.nml` and remove that file?

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Looks good. A few minor comments/questions.
We'll help you out on the travis-ci front if the simple fix that I suggested is not sufficient.

File: test/Data/72x35x25/input_bgc.nml (hunk 1 above)
Comment (liuxiao37k) [reply test/Data/72x35x25/input_bgc.nml]: Yes! We could definitely just use a single input.nml with the additional info added. I just wasn't sure how much you are willing to modify input.nml given that we have to add a bunch lines. I'll make the changes accordingly. Thanks.

**Comment by guillaumevernieres** [comment]
@travissluka , could you have a look at the travis-ci issue?

**Review by travissluka — APPROVED** [review]
@guillaumevernieres  `-DENABLE_OCEAN_BGC=ON` needed to be added to the MOM6 repo as well. Alternatively the flag could have been added to just `BUILD_OPT=` which would then pass it to all repos, but then any repo that doesn't use it would throw an annoying warning.

**Review by guillaumevernieres — APPROVED** [review]

**Pull request merged**